geometric_kernels/spaces/hyperbolic.py

```diff
@@ -30,6 +30,9 @@ def __init__(self, dim=1):
     def dimension(self) -> int:
         return self.dim
 
+    def random(self, key, number):
+        raise NotImplementedError
```

Comment:
Random sampling on noncompact spaces is infeasible: there is no uniform probability measure. Not sure what is the best way to go about it, maybe only include the abstract method in a compact base class (discrete spectrum)